src/plugins/event-plugins/__tests__/JsErrorPlugin.test.ts

```diff
@@ -521,4 +521,37 @@ describe('JsErrorPlugin tests', () => {
             })
         );
     });
+
+    test('when errors are ignored then no errors are recorded', async () => {
+        // Init
+        const plugin: JsErrorPlugin = new JsErrorPlugin({
+            shouldIgnoreError: (errorEvent: ErrorEvent) => {
+                const patterns = [/ResizeObserver loop/, /undefined/];
+                return (
+                    patterns.filter((pattern) =>
+                        pattern.test(errorEvent.error.message)
+                    ).length !== 0
+                );
+            }
+        });
+
+        // Run
+        plugin.load(context);
+
+        plugin.record({
+            name: 'Error',
+            message: 'ResizeObserver loop limit exceeded',
+            fileName: 'main.js',
+            lineNumber: 1,
+            columnNumber: 2
+        });
+        // This creates an error event with 'undefined' as message
+        plugin.record({});
+        plugin.record({});
+        plugin.record({});
+        plugin.disable();
+
+        // Assert
+        expect(record).toHaveBeenCalledTimes(0);
```

Comment:
Also should have a test to make sure it is called a certain amount of times with a mix of errors